docs/proposals/scheduling/20230314-gpu-topology-aware-scheduling.md

```diff
@@ -0,0 +1,275 @@
+gpu_6  96.42   16.75   17.47   15.89   48.35   96.28   754.10  48.33
+gpu_7  15.65   96.20   16.77   15.71   96.25   48.38   48.33   754.83
+```
+And the CR example is described as below:
+```
+apiVersion: scheduling.koordinator.sh/v1alpha1
+kind: Device
+metadata:
+  name: host04
+spec:
+  # report the gpu topology information
+  topologies:
+    gpu: 
+    - name: bandwidth
+      topology: '[[750.48,48.39,48.39,96.46,15.97,16.15,96.41,16.18],[48.39,752.65,96.46,48.38,16.96,16.84,16.67,96.23],[48.38,96.25,749.04,6.02,48.38,17.57,17.54,16.95],[96.44,48.39,96.48,752.65,17.27,48.38,17.33,16.8],[15.99,16.8,48.38,17.27,755.56,96.44,48.38,96.44],[16.14,16.74,17.73,48.38,96.23,755.56,96.25,48.38],[96.43,16.81,17.6,17.35,48.33,96.23,754.83,48.39],[16.28,96.22,17.18,16.88,96.23,48.33,48.33,755.56]]'
+  devices:
+  - health: true
+    id: GPU-04cea5cd-966f-7116-1d58-1ac34421541b
+    minor: 0
+    resources:
+      kubernetes.io/gpu-core: "100"
+      kubernetes.io/gpu-memory: 16Gi
+      kubernetes.io/gpu-memory-ratio: "100"
+    type: gpu
+  - health: true
+    id: GPU-3680858f-1753-371e-3c1a-
+    minor: 1
+    resources:
+      kubernetes.io/gpu-core: "100"
+      kubernetes.io/gpu-memory: 16Gi
+      kubernetes.io/gpu-memory-ratio: "100"
+    type: gpu
+  - health: true
+    id: GPU-95fe7a8b-bf9b-73cc-2903-c6e65883f3a7
+    minor: 2
+    resources:
+      kubernetes.io/gpu-core: "100"
+      kubernetes.io/gpu-memory: 16Gi
+      kubernetes.io/gpu-memory-ratio: "100"
+    type: gpu
+  - health: true
+    id: GPU-cd8d5d8c-7334-4c68-587e-04202daa38a5
+    minor: 3
+    resources:
+      kubernetes.io/gpu-core: "100"
+      kubernetes.io/gpu-memory: 16Gi
+      kubernetes.io/gpu-memory-ratio: "100"
+    type: gpu
+  - health: true
+    id: GPU-511dd579-5044-b716-e08a-841f51796a59
+    minor: 4
+    resources:
+      kubernetes.io/gpu-core: "100"
+      kubernetes.io/gpu-memory: 16Gi
+      kubernetes.io/gpu-memory-ratio: "100"
+    type: gpu
+  - health: true
+    id: GPU-62460a09-6838-abc8-00f5-31d2c6c101ef
+    minor: 5
+    resources:
+      kubernetes.io/gpu-core: "100"
+      kubernetes.io/gpu-memory: 16Gi
+      kubernetes.io/gpu-memory-ratio: "100"
+    type: gpu
+  - health: true
+    id: GPU-2da27328-f395-a226-7486-c08e6e98570f
+    minor: 6
+    resources:
+      kubernetes.io/gpu-core: "100"
+      kubernetes.io/gpu-memory: 16Gi
+      kubernetes.io/gpu-memory-ratio: "100"
+    type: gpu
+  - health: true
+    id: GPU-8137b226-b69a-1f22-4367-da110c8ba6b5
+    minor: 7
+    resources:
+      kubernetes.io/gpu-core: "100"
+      kubernetes.io/gpu-memory: 16Gi
+      kubernetes.io/gpu-memory-ratio: "100"
+    type: gpu
+```
+#### Node Selection
+Suppose a training job has a total of 3 pods waiting to be scheduled, namely pod1, pod2, and pod3. When pod1 is scheduled, the list of available nodes filtered by the filter extension point is [node1, node2, node3]. The logic for selecting available node groups for [pod1, pod2, pod3] is as follows:
```

Comment:
in main steps part, all process happen in prescore\score step, filter step is not mentioned. but here you mention filter step, which seems strange.